### ISSUE TITLE
[Prototype] Installable date picker code

### ISSUE THREAD

**Issue by sophschneider** [Shopify/polaris-react-archive#8761]
We need to answer: **How will the pattern code be made installable?**

Outcomes:
1. Recommendation of where the pattern code should live
2. Add components (probably date picker patterns) under the recommended infrastructure in a branch
3. Feedback from the Polaris team on the prototype

**Comment by sophschneider** [comment]
High level exploration: https://www.figma.com/file/ugVGHJtKvFSIdmhvOfInjW/Patterns-v2-Ownership-research?node-id=0%3A1&t=3ebJTGtEauhl70Rt-1

We've now split up the branches to each investigate the viable solutions in depth. We will present our findings to each other at the end of the week.

**Comment by sophschneider** [comment]
- We made the decision to create a workspace under the Polaris monorepo [here](https://vault.shopify.io/gsd/projects/33833/decisions/1978).
- We created a prototype branch doing that [here](https://github.com/Shopify/polaris/pull/8819/files). 

To start off, I copied a lot of the infrastructure from `polaris-react`. Here I want to discuss any changes we want to make and how to properly create our `package.json` with appropriate dependencies. 

S/o to Charles for this table from the monorepo exploration that I'm stealing to outline this

| Infrastructure | Decision | Notes |
|--------|--------|--------|
| Unit testing | ✅ | Copy jest setup from `polaris-react`. I don't see any reason why this would be an issue, plus we can reuse the CI |
| Linting / prettier / code formatting | ✅ | monorepo linter configs should work  |
| Release/Versioning/Changelogs | ✅ | Use the monorepo's changeset setup. We should still tophat the release flow (run `/snapit`?) |
| Storybook/Chromatic | ❓ | Copy from `polaris-react` but update to 7.0 and use actual built `polaris-react` assets |
| Typescript | ❓ | Copy from tsconfig from `polaris-react` ? How can we make sure the typescript versions are compatible between the two? |
| polaris-react | ❓ | peer dependency or dependency? |
| Build |❓ | Copy rollup setup from `polaris-react`?  Should we centralize the scripts or keep them separated? |
| Documentation | ❓ | This should be addressed in https://github.com/Shopify/polaris/issues/8762 |
| Workspace Maintainer management | ❓ | How do we separate polaris-react issues from polaris-pattern issues? implement codeowners and make it a requirement to add a component? |
| CI | ❓ | copy over polaris-react workflows? |
| Contribution guidelines | ❓ | What should we write in them |

**Comment by sophschneider** [comment]
.

**Comment by sophschneider** [comment]
.

**Comment by sophschneider** [comment]
Closing this issue in light of our prototyping. The Date Picker components are highly impactful and will be beneficial additions to our library in the future, but it will slow down the completion of this project. The components require both string internationalization and a date management library, which is not easy to architect in a library. [MUI](https://mui.com/x/react-date-pickers/adapters-locale/) for example, lets the consumer pick a date management library and plug into their "adapter". In addition to localization concerns, the DateRangePicker component is extremely logic heavy and complicated. In order to ensure the success of this project and that we are shipping something useful quickly, we've decided to descope date picking from this project and focus on the two existing layout patterns (https://polaris.shopify.com/patterns/resource-details-layout, https://polaris.shopify.com/patterns/app-settings-layout) in our prototype. We will still be picking new patterns for our build phase in addition to these two layout patterns.

This way, we can focus on more simple compositional components and scale up the complexity of the library after an MVP is shipped.
